app/controllers/troubleshooting_controller.rb

```diff
@@ -0,0 +1,181 @@
+  # unfortunately we're dealing with tags, which have numerous subclasses.
+  def item_path
+    if @item.is_a?(Tag)
+      tag_path(@item)
+    else
+      polymorphic_path(@item)
+    end
+  end
+
+  # The path to view the troubleshooting page for this item. Again, this is
+  # necessary because tags have a lot of subclasses and need special handling.
+  def troubleshooting_path
+    if @item.is_a?(Tag)
+      tag_troubleshooting_path(@item)
+    else
+      polymorphic_path([@item, :troubleshooting])
+    end
+  end
+
+  # Load the @item based on params. Results in a 404 error if the item in
+  # question can't be found, and a 500 error if there's an unknown type. Also
+  # sets the variable @allowed_actions.
+  def load_item
+    if params[:tag_id]
+      @item = Tag.find_by_name(params[:tag_id])
+
+      if @item.nil?
+        raise ActiveRecord::RecordNotFound,
+              ts("Could not find tag with name '%{name}'",
+                 name: params[:tag_id])
+      end
+    elsif params[:work_id]
+      @item = Work.find(params[:work_id])
+    else
+      raise "Unknown item type!"
+    end
+
+    @check_visibility_of = @item
+    @allowed_actions = allowed_actions
+  end
+
+  ########################################
+  # AVAILABLE ACTIONS
+  ########################################
+
+  # An action allowing the user to reindex a work.
+  def reindex_work
+    @item.enqueue_to_index
+    flash[:notice] << ts("Work sent to be reindexed.")
+  end
+
+  # An action allowing the user to reindex a tag (and everything related to it).
+  def reindex_tag
+    @item.async(:reindex_associated, true)
+    flash[:notice] << ts("Tag reindex job added to queue.")
+  end
+
+  # An action allowing the user to try to fix the filters for this tag and all
+  # of its synonyms.
+  def update_tag_filters
+    @item.async(:update_filters_for_taggables)
+
+    @item.synonyms.find_each do |syn|
+      syn.async(:update_filters_for_taggables)
+    end
+
+    flash[:notice] << ts("Tagged items enqueued for filter updates.")
+  end
+
+  # An action allowing the user to try to fix a single work's filters.
+  def update_work_filters
+    @item.update_filters
+    flash[:notice] << ts("Work filters updated.")
+  end
+
+  # An action allowing the user to try to fix the filter count and taggings
+  # count.
+  def fix_counts
+    @item.filter_count&.update_counts
+    @item.update(taggings_count: @item.taggings.count)
+    flash[:notice] << ts("Tag counts updated.")
+  end
+
+  # An action allowing the user to try to fix the inherited meta tags.
+  def fix_meta_tags
+    modified = MetaTagging.transaction do
+      InheritedMetaTagUpdater.new(@item).update
+    end
+
+    if modified
+      # Fixing the meta taggings is all well and good, but unless the filters
+      # are adjusted too, this will have no immediate effects.
+      @item.async(:update_filters_for_filterables)
+      flash[:notice] << ts("Inherited meta tags recalculated. This tag has " \
```

Comment:
Could you change this, and the instances on lines 171 and 172, to the official spelling of "metatags"?

Reply:
The header on the tag page currently spells it as "Meta tags."
https://github.com/otwcode/otwarchive/blob/78bdfeefc003aa3e904f6ca66a9ad00f4557886b/app/views/tags/show.html.erb#L74
Are there some contexts where "meta tags" is preferred over "metatags," or should I change the header, too? Or should I restrict myself to fixing the spelling only on the lines that I've already touched? (Does that include test names and comments, or is it only the error messages that matter?) Is it okay to have an error message that spells it once as "meta tags" and once as "metatags"? That is, do these lines also need changing?
https://github.com/otwcode/otwarchive/blob/78bdfeefc003aa3e904f6ca66a9ad00f4557886b/app/models/tag.rb#L1182
https://github.com/otwcode/otwarchive/blob/78bdfeefc003aa3e904f6ca66a9ad00f4557886b/app/models/meta_tagging.rb#L10
(I think I'd also need to change it in the locales file, too, because the attribute name is used in that second error message, and it defaults to turning the underscore in `meta_tag_id` into a space.)

Reply:
"Metatags" is what's consistently used in documentation, so in an ideal world, it should be that everywhere. But unless you're itching to fix it everywhere, I'm happy to just stick with the lines you've touched, and really only worrying about user-facing stuff (e.g. error messages, not code comments or test names). If fixing all instances of "meta tags" in certain error messages you've touched will be a pain, we can live with a mix of "meta tags" and "metatags."

Reply:
I think I'll take a crack at changing it in more places. Should "sub tag" become "subtag" as well?

Reply:
Sounds good! And yeah, "subtag" is also preferred. Thank you so much!